test/Microsoft.TestPlatform.CommunicationUtilities.UnitTests/Serialization/TestResultSerializationTests.cs

```diff
@@ -212,45 +216,66 @@ public void TestResultObjectShouldContainAllPropertiesOnDeserializationV2()
         {
             var json = "{\"TestCase\":{\"Properties\":[{\"Key\":{\"Id\":\"TestCase.FullyQualifiedName\",\"Label\":\"FullyQualifiedName\",\"Category\":\"\",\"Description\":\"\",\"Attributes\":1,\"ValueType\":\"System.String\"},\"Value\":\"sampleTestClass.sampleTestCase\"},{\"Key\":{\"Id\":\"TestCase.ExecutorUri\",\"Label\":\"Executor Uri\",\"Category\":\"\",\"Description\":\"\",\"Attributes\":1,\"ValueType\":\"System.Uri\"},\"Value\":\"executor://sampleTestExecutor\"},{\"Key\":{\"Id\":\"TestCase.Source\",\"Label\":\"Source\",\"Category\":\"\",\"Description\":\"\",\"Attributes\":0,\"ValueType\":\"System.String\"},\"Value\":\"sampleTest.dll\"},{\"Key\":{\"Id\":\"TestCase.Id\",\"Label\":\"Id\",\"Category\":\"\",\"Description\":\"\",\"Attributes\":1,\"ValueType\":\"System.Guid\"},\"Value\":\"28e7a7ed-8fb9-05b7-5e90-4a8c52f32b5b\"}]},\"Attachments\":[{\"Uri\":\"http://dummyUri\",\"DisplayName\":\"sampleAttachment\",\"Attachments\":[]}],\"Messages\":[],\"Properties\":[]}";
 
-            var result = Deserialize<TestResult>(json, 2);
+            var result = Deserialize<TestResult>(json, version);
 
             Assert.AreEqual(1, result.Attachments.Count);
             Assert.AreEqual(new Uri("http://dummyUri"), result.Attachments[0].Uri);
             Assert.AreEqual("sampleAttachment", result.Attachments[0].DisplayName);
         }
 
         [TestMethod]
-        public void TestResultPropertiesShouldGetRegisteredAsPartOfDeserializationV2()
+        [DataRow(2)]
+        [DataRow(3)]
+        public void TestResultPropertiesShouldGetRegisteredAsPartOfDeserializationV2(int version)
         {
             TestProperty.TryUnregister("DummyProperty", out var property);
             var json = "{\"TestCase\":{\"Id\":\"28e7a7ed-8fb9-05b7-5e90-4a8c52f32b5b\",\"FullyQualifiedName\":\"sampleTestClass.sampleTestCase\",\"DisplayName\":\"sampleTestClass.sampleTestCase\",\"ExecutorUri\":\"executor://sampleTestExecutor\",\"Source\":\"sampleTest.dll\",\"CodeFilePath\":null,\"LineNumber\":-1,\"Properties\":[]},\"Attachments\":[],\"Outcome\":1,\"ErrorMessage\":\"sampleError\",\"ErrorStackTrace\":\"sampleStackTrace\",\"DisplayName\":\"sampleTestResult\",\"Messages\":[],\"ComputerName\":\"sampleComputerName\",\"Duration\":\"10675199.02:48:05.4775807\",\"StartTime\":\"2007-03-10T00:00:00+00:00\",\"EndTime\":\"9999-12-31T23:59:59.9999999+00:00\"," +
                 "\"Properties\":[{\"Key\":{\"Id\":\"DummyProperty\",\"Label\":\"DummyPropertyLabel\",\"Category\":\"\",\"Description\":\"\",\"Attributes\":5,\"ValueType\":\"System.String\"},\"Value\":\"dummyString\"},]}";
 
-            var test = Deserialize<TestResult>(json, 2);
+            var test = Deserialize<TestResult>(json, version);
 
             this.VerifyDummyPropertyIsRegistered();
         }
 
         #endregion
 
+        #region future
+
+        [TestMethod]
+        public void TestResultSerializationShouldThrowWhenAVersionOfProtocolThatDoesNotExistYetIsProvided()
```

Comment:
Could simplified to
`TestResultSerializationShouldThrowWhenProvidedProtocolVersionDoesNotExist`

Reply:
Thx. Consider making it a suggestion next time :)

![image](https://user-images.githubusercontent.com/5735905/99041908-0b4a4d00-258c-11eb-9f3e-92dee071738f.png)